test/clj/game_test/cards/operations.clj

```diff
@@ -180,6 +180,20 @@
     (is (= 1 (:click (get-corp))))
     (is (= 7 (count (:hand (get-corp)))) "Drew 2 cards")))
 
+(deftest building-blocks
+  ;; Building Blocks - install and rez a barrier from HQ at no cost
+  (do-game
+    (new-game (default-corp ["Building Blocks" "Ice Wall"])
+              (default-runner))
+    (core/gain state :corp :credit 1)
+    (play-from-hand state :corp "Building Blocks")
+    (is (= 1 (:credit (get-corp))) "Corp starts with 1 credit")
+    (prompt-select :corp (find-card "Ice Wall" (:hand (get-corp))))
+    (prompt-choice :corp "New remote")
+    (let [iw (get-ice state :remote1 0)]
+      (is (= 1 (:credit (get-corp))) "Corp spent no credits")
+      (is (:rezzed (refresh iw)) "Ice Wall is installed and rezzed"))))
```

Comment:
I'd also like a test where you try to select a non-Barrier, and maybe a non-Barrier, non-ice as well.